### PR TITLE
Move HTTP-specific `proxy` modules into proxy::http

### PR DESCRIPTION
**Pull request by olix0r**
To prepare to move http-specific proxying logic from bind.rs into
`proxy`, let's carve out a `proxy::http` module for HTTP-specific behavior.

### CODE REVIEW

**Comment by hawkw** [comment]
Based on the [build failure](https://travis-ci.org/linkerd/linkerd2-proxy/builds/428267662?utm_source=github_status&utm_medium=notification) it looks like there's an import of `proxy::h2_router` in `Inbound` that needs to be pointed at `proxy::http::router`?

File: src/proxy/http/router.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ where
     router: Router<R>,
 }
 
-pub struct H2Router<R>
+pub struct RouterService<R>
```
Comment (hawkw) [inline src/proxy/http/router.rs]: TIOLI: why not just `Service`, so it can be referred to as `router::Service`?

File: src/proxy/http/glue.rs [hunk 2]
```diff
@@ -35,13 +35,13 @@ pub enum HttpBody {
 
 /// Glue for `tower_h2::Body`s to be used in hyper.
 #[derive(Debug, Default)]
-pub(super) struct BodyPayload<B> {
+pub struct BodyPayload<B> {
```
Comment (hawkw) [inline src/proxy/http/glue.rs]: Do these need to be `pub`, or can they just be `pub(in proxy::http)` or something?

Comment (olix0r) [reply src/proxy/http/glue.rs]: they can probably be `pub(in proxy)` (i wasn't aware of the _in_ syntax...)

**Pull request merged**